### PR TITLE
Can only be installed in SLE12 platforms.

### PR DESCRIPTION
**Pull request by aplanas**
Depends on crowbar/barclamp-crowbar#1175

### CODE REVIEW

**Comment by jsuchome** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Comment by tboerger** [comment]
:+1:

**Comment by aplanas** [comment]
* Change the constraint to accept RHEL, Centos, and other distributions, except SLE11 (SLE12 is OK)

**Comment by tboerger** [comment]
Constraints like < 12, <= 12 should be interesting for things like that as everything below 12.0 is not supported.

**Comment by jsuchome** [comment]
Hm, are we sure about radosgw role? We used to put it on controller nodes...

File: crowbar_framework/app/models/ceph_service.rb [hunk 1]
```diff
@@ -48,15 +48,24 @@ def role_constraints
       {
         "ceph-mon" => {
           "unique" => false,
-          "count" => 3
+          "count" => 3,
+          "exclude_platform" => {
+            "suse" => "/11.*/"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb]: ```< 12.0```

File: crowbar_framework/app/models/ceph_service.rb [hunk 2]
```diff
@@ -48,15 +48,24 @@ def role_constraints
...
+          }
         },
         "ceph-osd" => {
           "unique" => false,
-          "count" => 8
+          "count" => 8,
+          "exclude_platform" => {
+            "suse" => "/11.*/"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb]: ```< 12.0```

File: crowbar_framework/app/models/ceph_service.rb [hunk 3]
```diff
@@ -48,15 +48,24 @@ def role_constraints
...
+          }
         },
         "ceph-radosgw" => {
           "unique" => false,
           "count" => 1,
+          "exclude_platform" => {
+            "suse" => "/11.*/"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb]: ```< 12.0```

**Comment by aplanas** [comment]
* Express the same constraint using '<' operator

**Comment by aplanas** [comment]
* Exclude Windows platforms

**Comment by jsuchome** [comment]
(regarding my previous comment: actually, ceph-radosgw role also includes the default roles and I assume it should install the same package version as other ceph roles, so yes, let's make it SLE12-only as well

**Pull request merged**